app/controllers/name_controller/create_and_edit_name.rb

```diff
@@ -139,32 +143,52 @@ def perform_merge_names(new_name)
       @name, new_name = new_name, @name
       old_display_name_for_log = @name[:display_name]
     end
+    @name.display_name = old_display_name_for_log
+
     # Fill in author if other has one.
     if new_name.author.blank? && @parse.author.present?
       new_name.change_author(@parse.author)
     end
     new_name.change_deprecated(change_deprecated) unless change_deprecated.nil?
-    @name.display_name = old_display_name_for_log
+
+    # move associations from @name to new_name and destroy @name
     new_name.merge(@name)
     args = { this: @name.real_search_name, that: new_name.real_search_name }
     flash_notice(:runtime_edit_name_merge_success.t(args))
+
     @name = new_name
+    if new_name.icn_id != old_identifier
```

Comment:
Should this also make sure old_identifier is not blank?  Do we want an email notification if the merge results in a missing ICN id being filled in?

Reply:
You're right. I will re-work the code so that there's no admin notification when merge results merely filling in a missing ICN id.